webpack/components/pf3Table/formatters/EntitlementsInlineEditFormatter.js

```diff
@@ -3,35 +3,34 @@ import { sprintf, translate as __ } from 'foremanReact/common/I18n';
 import { KEYCODES } from 'foremanReact/common/keyCodes';
 import { Table, FormControl, FormGroup, HelpBlock, Spinner } from 'patternfly-react';
 import { validateQuantity } from '../../../scenes/Subscriptions/SubscriptionValidations';
+import { getEntitlementsDisplayValue } from '../../../scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers';
 
-const renderValue = (value, additionalData, onActivate) => {
+const renderValue = (rawValue, additionalData, onActivate) => {
   const { available, upstream_pool_id: upstreamPoolId, collapsible } = additionalData.rowData;
 
-  if (collapsible) {
-    return (
-      <td>{__('NA')}</td>
-    );
-  } else if (available < 0 || !upstreamPoolId) {
-    return (
-      <td>{available < 0 ? __('Unlimited') : available}</td>
-    );
-  }
+  const value = getEntitlementsDisplayValue({
+    rawValue, available, collapsible, upstreamPoolId,
+  });
+  const editable = (value === rawValue);
```

Comment:
I'm not sure that this is the correct check.  :thinking:

Reply:
This check aside for a moment, here's what I believe the overall functionality of this formatter should be so you can validate against that:

- If there are multiple subscriptions (iirc we group them by the name) then show NA because the row is `collapsible` and will never be editable as it does not correspond to a single subscription
- If there is an `upstreamPoolId` for the row (it maps to some subscription in the portal) then it should be `editable`
- If the value of `rawValue` (I think that's the quantity added in the manifest for example?) is -1, then show Unlimited as the text. I think this only relates to:
   - custom products
   - subscriptions created 'on the fly' in local Candlepin 
   - In other words these probably shouldn't be editable

Reply:
All that said, it seems like this would work for `editable`
`const editable = !!upstreamPoolId;`

And a more defensive check could be:

`const editable = !!upstreamPoolId && rawValue != -1`

Because subscriptions on the portal are attached in some quantity or another.

And of course you should validate what I've told you here against your own understanding and testing! If this affects the existing test cases then those should be carefully reviewed

Reply:
It looked good to me, but failed the test "`renders unlimited for -1`" because it was making 'Unlimited' editable.  I changed the check to `typeof value === 'number'`.  This should cover both `NA` and `Unlimited`, and hopefully make `editable` correct.   let me know your thoughts.

Reply:
that makes sense to me. code looks good. will test on monday!